### PR TITLE
first iteration of the provider attributes schema

### PR DESCRIPTION
**Pull request by baktun14**
- Added first iteration of the list of proposed provider attributes
- The idea is to have a base to start working with and that is publicly visible/modifyable. From there, it will be easy to adapt and evolve based on what we learn using it.
- The purpose of putting it in a json format with key/value is to be able to use the schema in a static form using react-hook-form with dynamic properties/possible values based on the schema, which will change when this schema changes.
- We will also use this schema to analyze the providers with our indexer and write somewhere on Cloudmos if the provider is missing some properties or if it has unknown properties.
- The biggest debate was around the location regions. We decided to go with regions and sub-regions where needed based on the United Nations Geoscheme https://en.wikipedia.org/wiki/United_Nations_geoscheme. In the list of attributes, providers can add their country, city and utc to further filter when creating an SDL.

TODO
----
- [x] Readme.md with the schema in a readable table
- [x] Country codes asia
~- [ ] Possibly add aliases for keys?~ Maybe in the next iteration

### CODE REVIEW

File: provider-attributes.json [hunk 1]
```diff
@@ -0,0 +1,676 @@
...
+        "key": "mix",
+        "description": "Mix of the above"
+      }
+    ]
+  },
+  "hosting-provider": {
+    "key": "hosting-provider",
+    "type": "string",
+    "required": false,
+    "description": "Name of the datacenter, colo, pubic cloud etc where hosted."
```
Comment (Redm4x) [inline provider-attributes.json]:
```suggestion
    "description": "Name of the datacenter, colo, public cloud, etc where hosted."
```

File: provider-attributes.json [hunk 2]
```diff
@@ -0,0 +1,676 @@
...
+    "required": true,
+    "description": "CPU manufacturer.",
+    "values": [
+      {
+        "key": "intel",
+        "description": "Intel"
+      },
+      {
+        "key": "amd",
+        "description": "Amd"
```
Comment (Redm4x) [inline provider-attributes.json]:
```suggestion
        "description": "AMD"
```

File: provider-attributes.json [hunk 3]
```diff
@@ -0,0 +1,676 @@
...
+        "key": "nvme",
+        "description": "nvme"
+      }
+    ]
+  },
+  "hardware-disk-size": {
+    "key": "hardware-disk-size",
+    "type": "number",
+    "required": false,
+    "description": "Storage size."
```
Comment (Redm4x) [inline provider-attributes.json]: Should we specify units? (ex: Bytes)

File: provider-attributes.json [hunk 4]
```diff
@@ -0,0 +1,676 @@
...
+        "key": "ddr4",
+        "description": "ddr4"
+      }
+    ]
+  },
+  "hardware-memory-size": {
+    "key": "hardware-memory-size",
+    "type": "number",
+    "required": false,
+    "description": "Memory size."
```
Comment (Redm4x) [inline provider-attributes.json]: Units?

File: provider-attributes.json [hunk 5]
```diff
@@ -0,0 +1,676 @@
...
+    "required": true,
+    "description": "Email associated with the provider."
+  },
+  "organization": {
+    "key": "organization",
+    "type": "string",
+    "required": true,
+    "description": "Name of the organization that owns the provider."
+  },
+  "provider-website": {
```
Comment (Redm4x) [inline provider-attributes.json]: Is the `provider-` prefix necessary for the `website` and `status-page`? Why do we need a prefix here, but not on `email` or `organization`?

File: provider-attributes.json (hunk 3 above)
Comment (baktun14) [reply provider-attributes.json]: Yes that's a good idea

File: provider-attributes.json (hunk 5 above)
Comment (baktun14) [reply provider-attributes.json]: True it could be without the prefix

**Comment by baktun14** [comment]
Maybe add `multiple-option` as value type to be able to select multiple values for GPU models for example.

**Comment by baktun14** [comment]
We decided to remove the disk size attribute for memory/disk since it makes less sense and can be hard for providers to maintain this. Also, we keep track of the provider capacity through the status endpoint of the providers. It's better to keep track of the capabilities of the provider only and not mix in sizes or metrics.

**Review by Redm4x — APPROVED** [review]

**Comment by cloud-j-luna** [comment]
Great work on the schemas. What is the reasoning behind using the United Nations Geoscheme? (I do like it a lot)

**Comment by baktun14** [comment]
> Great work on the schemas. What is the reasoning behind using the United Nations Geoscheme? (I do like it a lot)

Thanks. I think it's the one that made most sense for us. It's a good standard and intuitive from a user standpoint and for providers to specify a region based on the continent/country and sub-region.

File: provider-attributes.json (hunk 3 above)
Comment (shimpa1) [reply provider-attributes.json]: I think we should specify the size and type ie hdd/ssd/nvme

File: provider-attributes.json (hunk 4 above)
Comment (shimpa1) [reply provider-attributes.json]: Units and type ie DDR3/DDR3ECC/DDR4/DDR4ECC/DDR5/DDR5ECC

File: provider-attributes.json (hunk 3 above)
Comment (baktun14) [reply provider-attributes.json]: > I think we should specify the size and type ie hdd/ssd/nvme

https://github.com/maxmaxlabs/cloudmos-config/pull/1#issuecomment-1653942001

File: provider-attributes.json (hunk 4 above)
Comment (baktun14) [reply provider-attributes.json]: I will add these types thanks.

**Pull request merged**